From a7bd568249f9ec20b69b2a700da6a0648e93a842 Mon Sep 17 00:00:00 2001
From: Marko Vasic <marko.z.vasic@gmail.com>
Date: Sat, 24 Sep 2016 14:32:50 -0500
Subject: [PATCH] [IO-483] FilenameUtils.getPrefixLength now works correctly
 for unix files/folder that are in the root folder and start with colons

---
 src/main/java/org/apache/commons/io/FilenameUtils.java        | 2 ++
 .../java/org/apache/commons/io/FilenameUtilsTestCase.java     | 4 ++++
 2 files changed, 6 insertions(+)

diff --git a/src/main/java/org/apache/commons/io/FilenameUtils.java b/src/main/java/org/apache/commons/io/FilenameUtils.java
index 3b1ef898717..b9b9c1848d0 100644
--- a/src/main/java/org/apache/commons/io/FilenameUtils.java
+++ b/src/main/java/org/apache/commons/io/FilenameUtils.java
@@ -667,6 +667,8 @@ public static int getPrefixLength(final String filename) {
                         return 2;
                     }
                     return 3;
+                } else if (ch0 == UNIX_SEPARATOR) {
+                    return 1;
                 }
                 return NOT_FOUND;
 
diff --git a/src/test/java/org/apache/commons/io/FilenameUtilsTestCase.java b/src/test/java/org/apache/commons/io/FilenameUtilsTestCase.java
index fc6a025f147..bd97a5ae2b9 100644
--- a/src/test/java/org/apache/commons/io/FilenameUtilsTestCase.java
+++ b/src/test/java/org/apache/commons/io/FilenameUtilsTestCase.java
@@ -583,6 +583,10 @@ public void testGetPrefixLength() {
         assertEquals(9, FilenameUtils.getPrefixLength("//server/a/b/c.txt"));
         assertEquals(-1, FilenameUtils.getPrefixLength("\\\\\\a\\b\\c.txt"));
         assertEquals(-1, FilenameUtils.getPrefixLength("///a/b/c.txt"));
+
+        assertEquals(1, FilenameUtils.getPrefixLength("/:foo"));
+        assertEquals(1, FilenameUtils.getPrefixLength("/:/"));
+        assertEquals(1, FilenameUtils.getPrefixLength("/:::::::.txt"));
     }
 
     @Test
